neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/core/internal/LazyStoreList.java

```diff
@@ -0,0 +1,416 @@
+        return eStore().lastIndexOf(owner, eStructuralFeature, object);
+    }
+
+    @Override
+    protected Object[] delegateToArray() {
+        return eStore().toArray(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected <T> T[] delegateToArray(T[] array) {
+        return eStore().toArray(owner, eStructuralFeature, array);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateGet(int index) {
+        return (E) eStore().get(owner, eStructuralFeature, index);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateSet(int index, E object) {
+        return (E) eStore().set(owner, eStructuralFeature, index, object);
+    }
+
+    @Override
+    protected void delegateAdd(Object object) {
+        delegateAdd(InternalEObject.EStore.NO_INDEX, object);
+    }
+
+    @Override
+    protected void delegateAdd(int index, Object object) {
+        eStore().add(owner, eStructuralFeature, index, object);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateRemove(int index) {
+        return (E) eStore().remove(owner, eStructuralFeature, index);
+    }
+
+    @Override
+    protected void delegateClear() {
+        eStore().clear(owner, eStructuralFeature);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected E delegateMove(int targetIndex, int sourceIndex) {
+        return (E) eStore().move(owner, eStructuralFeature, targetIndex, sourceIndex);
+    }
+
+    @Override
+    // FIXME Does not resolve
+    protected boolean delegateEquals(Object object) {
+        if (object == this) {
+            return true;
+        }
+
+        if (!List.class.isInstance(object)) {
+            return false;
+        }
+
+        List<?> list = List.class.cast(object);
+        return list.size() == delegateSize()
+                && list.equals(eStore().getAll(owner, eStructuralFeature));
+    }
+
+    @Override
+    protected int delegateHashCode() {
+        return eStore().hashCode(owner, eStructuralFeature);
+    }
+
+    @Override
+    protected String delegateToString() {
+        return eStore().getAll(owner, eStructuralFeature)
+                .stream()
+                .map(String::valueOf)
+                .collect(Collectors.joining(", ", "[", "]"));
+    }
+
+    @Override
+    protected Iterator<E> delegateIterator() {
+        return iterator();
+    }
+
+    @Override
+    protected ListIterator<E> delegateListIterator() {
+        return listIterator();
+    }
+
+    @Override
+    protected List<E> delegateBasicList() {
+        final Object[] data = eStore().toArray(owner, eStructuralFeature);
+
+        return data.length == 0
+                ? ECollections.emptyEList()
+                : new EcoreEList.UnmodifiableEList<>(owner, eStructuralFeature, data.length, data);
+    }
+
+    // endregion
+
+    // region Overrides from `DelegatingEcoreEList`
+
+    @Nonnull
+    @Override
+    // FIXME Does not resolve
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1134)